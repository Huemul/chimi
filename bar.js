console.log('\n----- bar package -----')
