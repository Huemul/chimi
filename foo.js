module.exports = () => {console.log('\n----- foo package -----')}
