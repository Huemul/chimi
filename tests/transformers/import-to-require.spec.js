const { stripIndent } = require('common-tags')

const importToRequire = require('../../lib/transformers/import-to-require')

describe('import-to-require', () => {
  it('should transform import to requires', () => {
    const inputCode = stripIndent`
      import foo from 'foo'

      import { bar } from 'bar'

      import * as qux from 'qux'

      console.log(foo)
      console.log(bar)
      console.log(qux)
    `

    const { code } = importToRequire()('README.md', inputCode)

    expect(code).toMatchSnapshot()
  })

  it('should handle invalid code', () => {
    const inputCode = stripIndent`
      import foo 'foo'

      console.log(foo)
    `

    const { code } = importToRequire({})('README.md', inputCode)

    expect(code).toBeNull()
  })
})
