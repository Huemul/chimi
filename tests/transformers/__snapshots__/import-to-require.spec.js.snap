// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-to-require should transform import to requires 1`] = `
"var _foo = require('foo');

var _bar = require('bar');

var _qux = require('qux');

console.log(_foo.default);
console.log(_bar.bar);
console.log(_qux);"
`;
